platform/android/java/lib/src/org/godotengine/godot/utils/PermissionsUtil.java

```diff
@@ -213,7 +213,7 @@ public static boolean requestManifestPermissions(Activity activity, @Nullable Se
 
 		List<String> manifestPermissions;
 		try {
-			manifestPermissions = getManifestPermissions(activity);
+			manifestPermissions = new ArrayList<String>(getManifestPermissions(activity));
```

Comment:
This is not the correct fix actually.
`getManifestPermissions` should not return `Collections.emptyList()` because it is immutable (and therefore, `remove` will throw an exception).

So in the `getManifestPermissions`, you should change the `return Collections.emptyList()` with `new ArrayList<>()`

Reply:
The empty list is not an issue here due to the check 
```
if (manifestPermissions.isEmpty()) {
			return true;
		}
```

I disagree about the fix being incorrect: wrapping Arrays.asList() in the implementation of getManifestPermissions instead of where it is called should behave equivalently.

I also do not understand the distinction between new ArrayList<>(Collections.emptyList()) which is what this code is effectively doing, and new ArrayList<>()

But anyway, changing the return type of getManifestPermissions to ArrayList, rather than abstract List could better illustrate that it is a mutable type. I would still be open to making this proposed change stylistically

Reply:
Yes, but the method is public. So other developers may call this method and do not expect it to be unmodifiable.

So we can decide between: 
1. `getManifestPermissions(..)` is always returning an unmodifiable list (like now) and we need to wrap it inside an `ArrayList` in order to remove permissions later
2. Make `getManifestPermissions` always return a mutable list

It probably does not matter which route we pick.
If it is the first, we should probably document this at `getManifestPermissions(..)`.
E.g. change `manifest permissions list` to `unmodifiable manifest permissions list`
and then we are good for the future :)

Reply:
@lyuma That's an interesting bug, thanks for finding it!

We're making modifications to the list returned by `getManifestPermissions(...)` within our own code, so I'd say let's be consistent and make `getManifestPermissions(...)` always return a mutable list as @Maran23 mentioned.